pkg/cloud/gcp/actuators/services/tags/tagservice.go

```diff
@@ -0,0 +1,40 @@
+package tagservice
+
+import (
+	"context"
+
+	tags "google.golang.org/api/cloudresourcemanager/v3"
+	"google.golang.org/api/option"
+)
+
+// TagService is a pass through wrapper for google.golang.org/api/cloudresourcemanager/v3
+// to enable tests to mock this struct and control behavior.
+type TagService interface {
+	GetNamespacedName(context.Context, string) (*tags.TagValue, error)
+}
+
+type tagService struct {
+	tagValuesService *tags.TagValuesService
+}
+
+// BuilderFuncType is function type for building gcp tag client.
+type BuilderFuncType func(ctx context.Context, serviceAccountJSON string) (TagService, error)
+
+// NewTagService return a new tagService.
+func NewTagService(ctx context.Context, serviceAccountJSON string) (TagService, error) {
+	service, err := tags.NewService(ctx, option.WithCredentialsJSON([]byte(serviceAccountJSON)))
+	if err != nil {
+		return nil, err
```

Comment:
Would appreciate wrapping this error please
```suggestion
		return nil, fmt.Errorf("could not create new tag service: %w", err)
```